Added image suggesting how to implement app2app-flow (#60)
vippsas/vipps-login-api: vipps-login-api.md

The Vipps app will return some data with the return to the `app_callback_uri`. 
It contains two query parameters `state` and `resume_uri`. 
`State` is the OIDC `state` parameter passed at the start of a login which can be used to identify the specific login if needed. 

The `resume_uri`parameter is generated by Vipps 
and can optionally be used to resume the login when the user returns from the Vipps app. 
If the merchant app manages to keep the initial browser window open it is not required to use the `resume_uri`. 
To use the `resume_uri` it must be opened in the same browser used in the initial phase 
(such as ASWebAuthenticationSession or Chrome Custom Tabs). It is required that the browser contains the cookies 
from the interaction that happens before the user is sent to the Vipps app.

Example `app_callback_uri` request
``` 
merchant-app://callback/?state=RFiQdrl_lvJUpVmTRSKmsZRGLM0G1N1qh0WebZ1gDNk&resume_uri=https%3A%2F%2Fapi.vipps.no%2Fvipps-login-idp%2Findex.html%3FtabId%3D7607f7f0-7ae2-49b7-9cb5-102143dac4ea
```

#### A typical flow/implementation might look like this:
![](images/app2appsuggestion.png)

> 1. An [OpenID authentication flow authorization URI](#oauth-20-authorize) is generated on the merchant backend. 

Reviewer: Could we maybe consistently start each sentence with what will perform the action? Might make it clearer who is  controlling each step.
```suggestion
1. Merchant backend generates an [OpenID authentication flow authorization URI](#oauth-20-authorize). 
```

   The URI is communicated to the merchant app

> 2. and is used to initiate Vipps login in an external browser that is opened within the app, 

Reviewer: My preference would be to start a new sentence
```suggestion
2. Merchant app uses the URI to initiate Vipps login in an external browser that is opened within the app, 
```

>    see [specification](#using-vipps-login-in-native-applications) for details.

Reviewer: ```suggestion
   see the [specification](#using-vipps-login-in-native-applications) for details.
```

> 3. If required, Vipps login will deeplink the user over to the Vipps app 

Reviewer: ```suggestion
3. Vipps login will open the Vipps app if required. 
```

   (if the user is remembered in the browser the user will be authenticated directly 
    - the user will then be on step 6 below)

> 4. When the user has approved the login in the Vipps app redirects the user to the `app_callback_uri`.

Reviewer: ```suggestion
4. Vipps app opens the deep link `app_callback_uri` parameter when the user approves the login.
```

> 5. The `app_callback_uri` should return the user to the merchant app. The app now has 2 separate choices.

Reviewer: ```suggestion
5. Merchant app receives the user. The app now has 2 separate choices.
```

   * Display the same browser instance that was created in 1.
   * Open a browser again using the `resume_uri`-query parameter that is returned with the request to the `app_callback_uri`.

> 6. Vipps login will finalise the authentication of the user and acquire consent to share information if needed. 

Reviewer: ```suggestion
6. Vipps login finalizes the authentication of the user and acquires consent to share information if needed. 
```

   When this is finished the user will be redirected to the `redirect_uri`. 
   The Vipps login process has now finished and the merchant controls the remaining process.

> 7. The merchant app sends the `code` and `state`parameters received in the callback to the merchant backend

Reviewer: Have to decide on "The merchant app" vs "Merchant app" if we go with the suggested pattern.
```suggestion
7. The merchant app sends the `code` and `state`parameters received in the callback to the merchant backend
```

#### A typical flow/implementation might look like this:
![](images/app2appsuggestion.png)
1. Merchant backend generates an [OpenID authentication flow authorization URI](#oauth-20-authorize). 
   The URI is communicated to the merchant app
2. Merchant app uses the URI to initiate Vipps login in an external browser that is opened within the app, 
   see the [specification](#using-vipps-login-in-native-applications) for details.
3. Vipps login will open the Vipps app if required.
   (if the user is remembered in the browser the user will be authenticated directly 
    - the user will then be on step 6 below)
4. Vipps app opens the deep link `app_callback_uri` parameter when the user approves the login.
5.  Merchant app receives the user. The app now has 2 separate choices.
   * Display the same browser instance that was created in 1.
   * Open a browser again using the `resume_uri`-query parameter that is returned with the request to the `app_callback_uri`.

> 6. Vipps login finalizes the authentication of the user and acquires consent to share information if needed. 

Reviewer: ```suggestion
6. Vipps login finalizes the authentication of the user and obtains consent to share information if needed. 
```
